heron/tools/cli/src/python/cli_helper.py

```diff
@@ -46,48 +47,49 @@ def create_parser(subparsers, action, help_arg):
 
 
 ################################################################################
-# pylint: disable=unused-argument
-def run(command, parser, cl_args, unknown_args, action):
+# pylint: disable=dangerous-default-value
+def run(command, cl_args, action, extra_args=[], extra_lib_jars=[]):
   '''
   helper function to take action on topologies
   :param command:
-  :param parser:
   :param cl_args:
-  :param unknown_args:
   :param action:        description of action taken
   :return:
   '''
-  try:
-    topology_name = cl_args['topology-name']
+  topology_name = cl_args['topology-name']
 
-    new_args = [
-        "--cluster", cl_args['cluster'],
-        "--role", cl_args['role'],
-        "--environment", cl_args['environ'],
-        "--heron_home", config.get_heron_dir(),
-        "--config_path", cl_args['config_path'],
-        "--override_config_file", cl_args['override_config_file'],
-        "--release_file", config.get_heron_release_file(),
-        "--topology_name", topology_name,
-        "--command", command,
-    ]
+  new_args = [
+      "--cluster", cl_args['cluster'],
+      "--role", cl_args['role'],
+      "--environment", cl_args['environ'],
+      "--heron_home", config.get_heron_dir(),
+      "--config_path", cl_args['config_path'],
+      "--override_config_file", cl_args['override_config_file'],
+      "--release_file", config.get_heron_release_file(),
+      "--topology_name", topology_name,
+      "--command", command,
+  ]
+  new_args += extra_args
 
-    if Log.getEffectiveLevel() == logging.DEBUG:
-      new_args.append("--verbose")
+  lib_jars = config.get_heron_libs(jars.scheduler_jars() + jars.statemgr_jars())
+  lib_jars += extra_lib_jars
 
-    lib_jars = config.get_heron_libs(jars.scheduler_jars() + jars.statemgr_jars())
+  if Log.getEffectiveLevel() == logging.DEBUG:
+    new_args.append("--verbose")
 
-    # invoke the runtime manager to kill the topology
-    execute.heron_class(
-        'com.twitter.heron.scheduler.RuntimeManagerMain',
-        lib_jars,
-        extra_jars=[],
-        args=new_args
-    )
+  # invoke the runtime manager to kill the topology
+  resp = execute.heron_class(
+      'com.twitter.heron.scheduler.RuntimeManagerMain',
+      lib_jars,
+      extra_jars=[],
+      args=new_args
+  )
 
-  except Exception:
-    Log.error('Failed to %s \'%s\'' % (action, topology_name))
-    return False
+  response.render(resp)
 
-  Log.info('Successfully executed %s \'%s\'' % (action, topology_name))
-  return True
+  if resp.status != response.Status.Ok:
```

Comment:
still some refactor left: push response object back to `main.py` so we can combine some logic.